examples/07_automatic_preprocessing_with_the_supervectorizer.py

```diff
@@ -0,0 +1,193 @@
+from sklearn.ensemble import HistGradientBoostingRegressor
+
+from sklearn.pipeline import Pipeline
+
+from dirty_cat import SuperVectorizer
+
+
+pipeline = Pipeline([
+    ('vectorizer', SuperVectorizer(auto_cast=True)),
+    ('clf', HistGradientBoostingRegressor(random_state=42))
+])
+
+###############################################################################
+# For reference, let's perform a cross-validation
+
+from sklearn.model_selection import cross_val_score
+
+
+scores = cross_val_score(pipeline, X, y, scoring='r2')
+
+print(f'{scores=}')
+print(f'mean={np.mean(scores)}')
+print(f'std={np.std(scores)}')
+
+###############################################################################
+# Compared to
+# :ref:`example 02<sphx_glr_auto_examples_02_fit_predict_plot_employee_salaries.py>`,
+# the mean score is a pretty much the same as the all-|SE|.
+
+###############################################################################
+# Analyzing what it does
+# ----------------------
+# Let's perform the same workflow, but without the `Pipeline`, so we can
+# analyze its mechanisms along the way.
+
+from sklearn.ensemble import RandomForestRegressor
+
+
+sup_vec = SuperVectorizer(auto_cast=True)
+regressor = RandomForestRegressor(n_estimators=25, random_state=42)
+# Fit the SuperVectorizer
+X_train_enc = sup_vec.fit_transform(X_train, y_train)
+X_test_enc = sup_vec.transform(X_test)
+# And the regressor
+regressor.fit(X_train_enc, y_train)
+
+###############################################################################
+# Inspecting the features created
+# -------------------------------
+# Once it has been trained on data,
+# we can print the transformers and the columns assignment it creates:
+
+print(sup_vec.transformers_)
+
+###############################################################################
+# This is what is being passed to the |ColumnTransformer| under the hood.
+# If you're familiar with how the later works, it should be very intuitive.
+# We can notice it classified the columns "gender" and "assignment_category"
+# as low cardinality string variables.
+# A |OneHotEncoder| will be applied to these columns.
+#
+# The vectorizer actually makes the difference between string variables
+# (data type ``object`` and ``string``) and categorical variables
+# (data type ``category``).
+#
+# Next, we can have a look at the encoded feature names.
+
+# Before:
+print(X.columns.to_list())
+# After :
+feature_names = sup_vec.get_feature_names()
+print(feature_names[:8])
+print(len(feature_names))
+
+###############################################################################
+# As we can see, it created a new column for each unique value.
+# This is because we used |SimilarityEncoder| on the column "division",
+# which was classified as a high cardinality string variable.
+# (default values, see `SuperVectorizer`'s docstring).
+# In total, we have 1212 encoded columns.
+#
```

Comment:
Here we could move the import and fit of the random forest, in a section call "Feature importance in the statistical model"

Reply:
Addressed in 4f7585e